lnrpc/routerrpc/router_backend.go

```diff
@@ -169,6 +170,26 @@ func (r *RouterBackend) QueryRoutes(ctx context.Context,
 		ignoredPairs[pair] = struct{}{}
 	}
 
+	// Since QueryRoutes allows having a different source other than
+	// ourselves, we'll only apply our max time lock if we are the source.
+	maxTotalTimelock := r.MaxTotalTimelock
+	if sourcePubKey != r.SelfNode {
+		maxTotalTimelock = math.MaxUint32
+	}
+	cltvLimit, err := ValidateCLTVLimit(in.CltvLimit, maxTotalTimelock)
+	if err != nil {
+		return nil, err
+	}
+
+	// We need to subtract the final delta before passing it into path
+	// finding. The optimal path is independent of the final cltv delta and
+	// the path finding algorithm is unaware of this value.
+	finalCLTVDelta := uint16(zpay32.DefaultFinalCLTVDelta)
+	if in.FinalCltvDelta != 0 {
+		finalCLTVDelta = uint16(in.FinalCltvDelta)
+	}
+	cltvLimit -= uint32(finalCLTVDelta)
```

Comment:
My thought was to do this subtracting in `ChannelRouter.FindRoute`. To me it sounds more logical to accept a `cltvLimit` for the whole route as a parameter to `FindRoute`.

Reply:
The godoc of the field in the `RestrictParams` struct indicates that it should not include the final CLTV delta. Are you suggesting we change this as well?

Reply:
Oh.. yes, that is indeed the interface towards `findPath`, because `findPath` knows nothing about final cltv delta. Maybe it would have been better if `FindRoute` had a different interface that doesn't expose this implementation detail. 

Probably leave it as it is for now.